### PR TITLE
Unicorn: add version 2.0.0

### PR DESCRIPTION
**Pull request by staskau**
Specify library name and version:  **unicorn/2.0.0**

Adding a version 2.0.0 of the Unicorn emulator.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch [hunk 1]
```diff
@@ -0,0 +1,9 @@
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -1416,7 +1416,7 @@ if(UNICORN_BUILD_TESTS)
+ endif()
+ 
+ 
+-if(UNICORN_INSTALL AND NOT MSVC)
++if(UNICORN_INSTALL)
+     include("GNUInstallDirs")
```
Comment (madebr) [inline recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch:9]: Needs a newline at the end.
Also, the patch is incomplete. There need to be 7 lines.

File: recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+     file(GLOB UNICORN_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/unicorn/*.h)
+     if (BUILD_SHARED_LIBS)
```
Comment (madebr) [inline recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch:11]: Looks like the install target is only available for shared libraries
```suggestion
@@ -1416,10 +1416,10 @@ if(UNICORN_BUILD_TESTS)
 endif()
 
 
-if(UNICORN_INSTALL AND NOT MSVC)
+if(UNICORN_INSTALL)
     include("GNUInstallDirs")
     file(GLOB UNICORN_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/unicorn/*.h)
-    if (BUILD_SHARED_LIBS)
+    if (1)
         install(TARGETS unicorn
             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
```

File: recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+--- CMakeLists.txt
```
Comment (madebr) [inline recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch:1]: The filename of this patch does not agree with the one in `conandata.yml`.
There is an extra dot after the `2.0.0`.

I wonder why conan/the hooks haven't complained.

File: recipes/unicorn/all/conandata.yml [hunk 4]
```diff
@@ -2,7 +2,13 @@ sources:
...
     url: "https://github.com/unicorn-engine/unicorn/archive/refs/tags/1.0.3.tar.gz"
     sha256: "64fba177dec64baf3f11c046fbb70e91483e029793ec6a3e43b028ef14dc0d65"
+  "2.0.0":
+    url: "https://github.com/unicorn-engine/unicorn/archive/refs/tags/2.0.0.tar.gz"
+    sha256: "67b445c760e2bbac663e8c8bc410e43311c7fc92df4dfa8d90e06a021d07f634"
 patches:
   "1.0.3":
     - patch_file: "patches/1.0.3-0001-cmake-crossbuild-support.patch"
       base_path: "source_subfolder"
+  "2.0.0.":
```
Comment (madebr) [inline recipes/unicorn/all/conandata.yml:12]: This is why the thooks haven't complained.
You weren't applying the patches
```suggestion
  "2.0.0":
```

File: recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch (hunk 3 above)
Comment (staskau) [reply recipes/unicorn/all/patches/2.0.0.-0001-cmake-msvc-support.patch]: You are absolutely right. Thank you vey much for your help!

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**